### ISSUE TITLE
perf: remove per-token wait_until_completed() in Metal forward pass

### ISSUE THREAD

**Issue by sauravpanda** [sauravpanda/forge-llm#155]
## Problem

Every GPU operation in the Metal forward pass is immediately followed by `cmd.wait_until_completed()`, causing **420+ CPU-GPU synchronization waits per token**. This serializes the entire compute graph on the CPU.

Current pattern in generated `model.rs`:
```rust
cmd.commit();
cmd.wait_until_completed();  // Blocks CPU after EVERY kernel
```

This occurs for embedding blits, KV cache updates, RMS norms, matmuls, RoPE, attention, SiLU, and residual adds — ~420 waits per token.

## Impact

**Estimated 50-60x slowdown.** Even if each kernel takes only 1ms, 420 synchronous waits = 420ms per token. llama.cpp runs kernels asynchronously and batches work into a single command buffer.

## Fix

1. Create a **single command buffer per forward pass** (not per operation)
2. Encode all operations sequentially into that buffer
3. Only call `wait_until_completed()` once at the end when reading back logits
4. Use `MTLEvent` or `encodeWait` for any genuine data dependencies

## Expected improvement

8 tok/s → ~400-500 tok/s (combined with other Metal optimizations)

## Files

- Codegen: `crates/forgellm-codegen-metal/src/lib.rs`
- Generated: `src/model.rs` lines 255, 284, 325, 358, 384, 408, 435, 455, 481
